pilot/pkg/bootstrap/servicecontroller.go

```diff
@@ -95,7 +97,18 @@ func (s *Server) initKubeRegistry(serviceControllers *aggregate.Controller, args
 	} else {
 		args.RegistryOptions.KubeOptions.EndpointMode = kubecontroller.EndpointsOnly
 	}
-	kubeRegistry := kubecontroller.NewController(s.kubeClient, s.metadataClient, args.RegistryOptions.KubeOptions)
+
+	kubeRegistry := kubecontroller.NewController(
+		s.kubeClients.Kube(),
+		s.kubeClients.KubeInformer().Core().V1().Pods(),
+		s.kubeClients.KubeInformer().Core().V1().Nodes(),
+		s.kubeClients.KubeInformer().Core().V1().Services(),
+		s.kubeClients.KubeInformer().Core().V1().Endpoints(),
+		s.kubeClients.KubeInformer().Discovery().V1alpha1().EndpointSlices(),
+		s.kubeClients.MetadataInformer().ForResource(corev1.SchemeGroupVersion.WithResource("nodes")),
```

Comment:
we dont need this right?

Reply:
Yeah,  I don't want to change it in this PR though, its big enough I can do it before in a split off PR or after